src/NServiceBus.Persistence.AzureStorage/SagaPersisters/AzureSagaPersister.cs

```diff
@@ -56,7 +56,11 @@ public Task Update(IContainSagaData sagaData, SynchronizedStorageSession session
                 EntityProperty value;
                 if (tableEntity.TryGetValue(SecondaryIndexIndicatorProperty, out value))
                 {
-                    meta.AddSecondaryIndexId(entity, PartitionRowKeyTuple.Parse(value.StringValue));
+                    var partitionRowKeyTuple = PartitionRowKeyTuple.Parse(value.StringValue);
+                    if (partitionRowKeyTuple.HasValue)
```

Comment:
In my opinion it can't be null as it is set in the following line
```
 toPersist.Add(SecondaryIndexIndicatorProperty, secondaryIndexKey.ToString());
```
This means, that if a value exists it will be a stringified secondaryIndexKey, which means that it won't be null. So this is not a bug fix 

@SeanFeldman : Having this ☝️ could you clarify your opinion on this PR?